engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/CallActivityTest.java

```diff
@@ -279,6 +279,67 @@ public void shouldApplyInputMappings() {
         .contains(tuple("x", "1"), tuple("y", "1"));
   }
 
+  @Test
+  public void shouldOnlyPropagateVariablesDefinedViaInputMappings() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            "wf-parent.bpmn",
+            parentProcess(
+                c -> c.zeebeInputExpression("x", "y").zeebePropagateAllParentVariables(false)))
+        .deploy();
+
+    // when
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID_PARENT)
+            .withVariables(Map.of("x", 1))
+            .create();
+
+    // then
+    final var childInstance = getChildInstanceOf(processInstanceKey);
+
+    assertThat(
+            RecordingExporter.variableRecords()
+                .withProcessInstanceKey(childInstance.getProcessInstanceKey())
+                .limit(2))
```

Comment:
🔧 We limit for 2, but there's only 1 variable record written for this process instance. As a result the test takes >5 seconds to execute. We should limit it to 1 instead.

Reply:
If you don't want to limit on 1 record because you might miss more records we'd have to get more creative with the filter. The child instance has a wait-state because of the service task. If we get all records, limit it to the Job.CREATED intent and then get our variables we make sure we get all of the variable created records.

```java
    assertThat(
            RecordingExporter.records()
                .limit(r -> r.getIntent() == JobIntent.CREATED)
                .variableRecords()
                .withIntent(VariableIntent.CREATED)
                .withProcessInstanceKey(childInstance.getProcessInstanceKey()))
        .extracting(Record::getValue)
        .extracting(v -> tuple(v.getName(), v.getValue()))
        .containsExactly(tuple("y", "1"));
```